slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/PruningIntegrationTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import tech.pegasys.web3signer.slashingprotection.dao.SignedAttestation;
+import tech.pegasys.web3signer.slashingprotection.dao.SignedBlock;
+import tech.pegasys.web3signer.slashingprotection.dao.SigningWatermark;
+
+import java.util.List;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class PruningIntegrationTest extends IntegrationTestBase {
+
+  @ParameterizedTest
+  @CsvSource({"1, 9", "2, 8", "5, 5", "9, 1", "10, 0", "20, 0"})
+  void prunesDataForRegisteredValidator(final int amountToKeep, final int pruningMark) {
+    final int size = 10;
+    insertAndRegisterData(size, size, 1);
+    final List<SignedAttestation> allAttestations = getAttestations(1);
+    final List<SignedBlock> allBlocks = getBlocks(1);
+
+    slashingProtection.prune(amountToKeep, 1);
+
+    final List<SignedAttestation> expectedAttestations = allAttestations.subList(pruningMark, size);
+    final List<SignedAttestation> attestations = getAttestations(1);
+    assertThat(attestations).usingFieldByFieldElementComparator().isEqualTo(expectedAttestations);
+
+    final List<SignedBlock> expectedBlocks = allBlocks.subList(pruningMark, size);
+    final List<SignedBlock> blocks = getBlocks(1);
+    assertThat(blocks).usingFieldByFieldElementComparator().isEqualTo(expectedBlocks);
+
+    final UInt64 expectedWatermarkValue = UInt64.valueOf(pruningMark);
+    assertThat(getWatermark(1))
```

Comment:
not sure why the 'getX' functions take a parameter, what does it mean? i.e. is it the validatorId? If so - could we call that out as a VALIDATOR_ID variable/constant?